### PR TITLE
[CI] use fixture for squad dataset tests

### LINKED ISSUE #697: Flaky test: test_load_vocabs
http://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-cpu-unittest/detail/PR-692/7/pipeline/81/ 

```
=================================== FAILURES ===================================

_______________________________ test_load_vocabs _______________________________

[gw3] linux -- Python 3.6.7 /var/lib/jenkins/workspace/gluon-nlp-cpu-py3@4/conda/cpu/py3/bin/python



    @pytest.mark.remote_required

    def test_load_vocabs():

>       dataset = SQuAD(segment='dev', root='tests/data/squad')



scripts/tests/test_question_answering.py:70: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

src/gluonnlp/data/question_answering.py:136: in __init__

    super(SQuAD, self).__init__(self._read_data())

src/gluonnlp/data/question_answering.py:174: in _read_data

    samples = json.load(f)

conda/cpu/py3/lib/python3.6/json/__init__.py:299: in load

    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)

conda/cpu/py3/lib/python3.6/json/__init__.py:354: in loads

    return _default_decoder.decode(s)

conda/cpu/py3/lib/python3.6/json/decoder.py:339: in decode

    obj, end = self.raw_decode(s, idx=_w(s, 0).end())

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 



self = <json.decoder.JSONDecoder object at 0x7f64a4946208>

s = '{"data": [{"title": "Super_Bowl_50", "paragraphs": [{"context": "Super Bowl 50 was an American football game to deter...rkers in the poor countries"}, {"answer_start": 239, "text": "low-skilled workers in the poor countries"}, {"answer_st'

idx = 0



    def raw_decode(self, s, idx=0):

        """Decode a JSON document from ``s`` (a ``str`` beginning with

        a JSON document) and return a 2-tuple of the Python

        representation and the index in ``s`` where the document ended.

    

        This can be used to decode a JSON document from a string that may

        have extraneous data at the end.

    

        """

        try:

>           obj, end = self.scan_once(s, idx)

E           json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 3342327 (char 3342326)



conda/cpu/py3/lib/python3.6/json/decoder.py:355: JSONDecodeError


```

### PR DESCRIPTION
**Pull request by szha**
## Description ##
use fixture for squad dataset tests. resolves #697 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] use fixture for squad dataset tests

### CODE REVIEW

File: scripts/tests/test_question_answering.py [hunk 1]
```diff
@@ -21,17 +21,15 @@
...
 
 question_max_length = 30
 context_max_length = 256
 
 
 @pytest.mark.remote_required
-def test_transform_to_nd_array():
-    dataset = SQuAD(segment='dev', version='1.1', root='tests/data/squad')
-    vocab_provider = VocabProvider(dataset)
+def test_transform_to_nd_array(squad_dev_and_vocab_provider):
```
Comment (eric-haibin-lin) [inline scripts/tests/test_question_answering.py]: why is `squad_dev_and_vocab_provider` an argument?

Comment (szha) [reply scripts/tests/test_question_answering.py]: it loads from the fixture

**Comment by mli** [comment]
Job PR-699/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-699/4/index.html

**Comment by mli** [comment]
Job PR-699/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-699/5/index.html

**Pull request merged**